### PR TITLE
Added TypeScript to the Language Section

### LINKED ISSUE #622: [Add] TypeScript in Languages section
### Site URL

https://www.freecodecamp.org/news/learn-typescript-beginners-guide/

### Description of the above provided site

- The link provided (https://www.freecodecamp.org/news/learn-typescript-beginners-guide/) leads to an article titled "Learn TypeScript: A Beginner's Guide" on the FreeCodeCamp website.
- This issue aims to highlight the advantages of TypeScript as a valuable resource for software development. TypeScript, a statically typed superset of JavaScript, brings several benefits to the table, enhancing code quality, maintainability, and developer productivity. By leveraging these advantages, developers can create robust and scalable applications.

### Category

Languages

### Sub Category

TypeScript

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by abhijeetnishal**
## Fixes Issue
Closes #622

## Changes proposed
1. Added TypeScript as a prominent language in the language section.
2. Added a list of curated resources for learning TypeScript, including official documentation, and community-driven learning platforms.
3. Ensured that the resource links are up-to-date and accessible to provide the most relevant and reliable learning material for developers.

## Screenshots
Screenshot 2023-05-23 at 11 48 11 PM

### CODE REVIEW

File: database/languages/typescript.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[
+    {
+        "name": "Beginners Guide",
```
Comment (CBID2) [inline database/languages/typescript.json:3]: [Typo]
```suggestion
        "name": "Typescript Beginners Guide",
```

File: database/languages/typescript.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        "description": "This is a comprehensive and accessible resource designed to introduce beginners to the fundamentals of TypeScript programming.",
```
Comment (CBID2) [inline database/languages/typescript.json:4]: [Typo]
```suggestion
        "description": "This is a comprehensive and accessible resource designed to introduce beginners to the fundamentals of TypeScript.",
```

File: database/languages/typescript.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        "url": "https://www.freecodecamp.org/news/learn-typescript-beginners-guide/",
+        "category": "Languages",
+        "subcategory": "typescript"
+    },
+    {
+        "name": "TypeScript Official",
```
Comment (CBID2) [inline database/languages/typescript.json:10]: [Typo]
```suggestion
        "name": "Official TypeScript Guide",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @abhijeetnishal! :) I pointed out some typos in my inline comments. Fix those and you're good to go! :)

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @abhijeetnishal! :) I pointed out some typos that could be improved. Fix those and you're good to go! :)

**Comment by abhijeetnishal** [comment]
Hey @CBID2 i have fixed the typos you have described.

**Review by CBID2 — APPROVED** [review]
Well done @abhijeetnishal! :) LGTM! :)

File: database/languages/typescript.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+[
+  {
+    "name": "TypeScript Beginners Guide",
+    "description": "This is a comprehensive and accessible resource designed to introduce beginners to the fundamentals of TypeScript.",
+    "url": "https://www.freecodecamp.org/news/learn-typescript-beginners-guide/",
+    "category": "Languages",
```
Comment (rupali-codes) [inline database/languages/typescript.json:6]:
```suggestion
    "category": "languages",
```

File: database/languages/typescript.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "subcategory": "typescript"
+  },
+  {
+    "name": "Official TypeScript Guide",
+    "description": "This is an official resource created by Microsoft to introduce TypeScript programming.",
+    "url": "https://www.typescriptlang.org/",
+    "category": "Languages",
```
Comment (rupali-codes) [inline database/languages/typescript.json:13]:
```suggestion
    "category": "languages",
```

**Comment by abhijeetnishal** [comment]
Hey @rupali-codes i have fixed the category typos you have described.

**Review by rupali-codes — APPROVED** [review]
lgtm :)

**Pull request merged**